File: Unidad8/SignalR/ChatClean/clsMensajeUsuario.cs
using System.Text.Json.Serialization;

namespace ChatClean.Models
{
    public class clsMensajeUsuario
    {
        [JsonPropertyName("usuario")]
        public string usuario { get; set; }

        [JsonPropertyName("mensaje")]
        public string mensaje { get; set; }

        public clsMensajeUsuario()
        {
            this.usuario = string.Empty;
            this.mensaje = string.Empty;
        }

        public clsMensajeUsuario(string usuario, string mensaje)
        {
            this.usuario = usuario;
            this.mensaje = mensaje;
        }

        public bool IsValid()
        {
            return !string.IsNullOrWhiteSpace(this.mensaje);
        }
    }
}

File: Unidad8/SignalR/ChatClean/clsMensajeUsuario.js
/**
 * Clase clsMensajeUsuario en JavaScript
 */
class clsMensajeUsuario {
    constructor(usuario = '', mensaje = '') {
        this.usuario = usuario;
        this.mensaje = mensaje;
    }

    static fromJSON(json) {
        return new clsMensajeUsuario(
            json.usuario || '',
            json.mensaje || ''
        );
    }

    toJSON() {
        return {
            usuario: this.usuario,
            mensaje: this.mensaje
        };
    }

    isValid() {
        return this.mensaje.trim().length > 0;
    }
}

// Log para verificar que se cargó
console.log('✅ clsMensajeUsuario.js cargado correctamente');